backend/bitsurance/bitsurance.go

```diff
@@ -0,0 +1,156 @@
+)
+
+// fetchBitsuranceIds fetches the list of insured account hashes from the Bitsurance
+// server. The result is returned as a slice of strings, with each element containing
+// the salted hash of an account xpub, in lexical order. (0-9a-z).
+// For details on the hashing please refer to `BitsuranceGetId`.
+func fetchBitsuranceIds(httpClient *http.Client) ([]string, error) {
+	endpoint := apiURL + "allAccountHashes"
+	req, err := http.NewRequest(http.MethodGet, endpoint, nil)
+	if err != nil {
+		return nil, errp.WithStack(err)
+	}
+	req.Header.Add("X-API-KEY", apiKey)
+
+	res, err := httpClient.Do(req)
+	if err != nil {
+		return nil, errp.WithStack(err)
+	}
+	defer res.Body.Close() //nolint:errcheck
+	if res.StatusCode != http.StatusOK {
+		return nil, errp.Newf("%s - bad response code %d", endpoint, res.StatusCode)
+	}
+	const max = 81920
+	responseBody, err := io.ReadAll(io.LimitReader(res.Body, max+1))
+	if err != nil {
+		return nil, errp.WithStack(err)
+	}
+	return strings.Split(string(responseBody), "\n"), nil
+}
+
+// findInsuredAccounts compares `accountIds` and `insuredIds` and returns a slice
+// with the matching elements.
+//
+// `accountIds` is an ordered slice of strings, containing the hashes of the
+// requested wallet accounts xpubs
+// `insuredIds` is another ordered slice of strings, containing the hashes of
+// all the insured xpubs retrieved from the Bitsurance server.
+//
+// The complexity of the scanning is linear.
+// For details on the hashing please refer to `BitsuranceGetId`.
+// TODO add tests.
+func findInsuredAccounts(accountIds, insuredIds []string) []string {
+	var matching []string
+
+	// Initialize indices for both lists
+	i, j := 0, 0
+
+	// Traverse both lists while there are elements in both
+	for i < len(accountIds) && j < len(insuredIds) {
+		// Compare the current elements in both lists
+		switch {
+		case accountIds[i] == insuredIds[j]:
+			// If the elements match, add it to the matching slice
+			matching = append(matching, accountIds[i])
+			i++
+			j++
+		case accountIds[i] < insuredIds[j]:
+			// If the element in accountIds is smaller, move to the next element
+			i++
+		default:
+			// If the element in insuredIds is smaller, move to the next element
+			j++
+		}
+	}
+
+	return matching
+}
+
+// BitsuranceGetId returns the BitsuranceId of a given account.
+// The id is computed hashing with sha256 the P2WPKH xpub of the account, concatenated with a fixed salt.
+// If a P2WPKH xpub can't be found in the account, empty string is returned.
+func BitsuranceGetId(account accounts.Interface) string {
+	for _, signingConf := range account.Info().SigningConfigurations {
+		bitcoinScriptType := signingConf.BitcoinSimple
+		if bitcoinScriptType != nil && bitcoinScriptType.ScriptType == signing.ScriptTypeP2WPKH {
+			h := sha256.New()
+			h.Write([]byte(bitcoinScriptType.KeyInfo.ExtendedPublicKey.String() + xpubSalt))
+			return hex.EncodeToString(h.Sum(nil))
+		}
+	}
+	return ""
```

Comment:
Why return "" and not an error? seems safer to return an error and force checking it at the call-site (that check is currently missing there)

Reply:
Added error handling